lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -279,36 +304,28 @@ class BlockedToMMA : public mlir::RewritePattern {
     }
     auto newRetType =
         RankedTensorType::get(retShape, oldRetType.getElementType(), mmaEnc);
-
     // convert accumulator
     auto oldAcc = dotOp.getOperand(2);
     auto newAcc = rewriter.create<triton::gpu::ConvertLayoutOp>(
         oldAcc.getLoc(), newRetType, oldAcc);
-    auto oldAOrder = oldAType.getEncoding()
-                         .cast<triton::gpu::DotOperandEncodingAttr>()
-                         .getParent()
-                         .cast<triton::gpu::BlockedEncodingAttr>()
-                         .getOrder();
-    auto oldBOrder = oldBType.getEncoding()
-                         .cast<triton::gpu::DotOperandEncodingAttr>()
-                         .getParent()
-                         .cast<triton::gpu::BlockedEncodingAttr>()
-                         .getOrder();
-
+    // convert operands
+    int minBitwidth = std::min(computeOrigBitWidth(a), computeOrigBitWidth(b));
+    Type minType = IntegerType::get(ctx, minBitwidth);
+    // convert A operand
     auto newAEncoding = triton::gpu::DotOperandEncodingAttr::get(
         oldAType.getContext(), 0, newRetType.getEncoding(),
-        oldAType.getElementType());
-    auto newBEncoding = triton::gpu::DotOperandEncodingAttr::get(
-        oldBType.getContext(), 1, newRetType.getEncoding(),
-        oldBType.getElementType());
-
+        minBitwidth > 0 ? minType : oldAType.getElementType());
```

Comment:
With this change getting into our fork, we will be using the [other constructor](https://github.com/ROCmSoftwarePlatform/triton/blob/9cdf3a58c396a8265a788d4b5a5672d9b0c0c22a/include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td#L642C16-L642C39) for dotOperand, which sets `kWidth` explicitly as 32 / elemBitWidth.
@binarman Can you double check if this is the case? If so, we'll probably need some fixes there.

Reply:
This function work only for MMA layout, so it should not affect MFMA layout.
This change is safe.

Reply:
In current triton-mlir - nowhere =)
I've added it in https://github.com/ROCmSoftwarePlatform/triton/pull/295/files#diff-980bbab8870c6a2f27dbee8696cea2a5d4db63de32bf63f6584ac1fbd3409d9fR209, it is not merged yet

Reply:
yes, we can revisit in #295 then once this one is merged.